AsyncDisplayKit/ASControlNode.mm

```diff
@@ -354,44 +345,28 @@ - (void)removeTarget:(id)target action:(SEL)action forControlEvents:(ASControlNo
     {
       // Grab the dispatch table for this event (if we have it).
       id<NSCopying> eventKey = _ASControlNodeEventKeyForControlEvent(controlEvent);
-      NSMapTable *eventDispatchTable = _controlEventDispatchTable[eventKey];
-      if (!eventDispatchTable)
+      NSMutableArray *eventTargetActionArray = _controlEventDispatchTable[eventKey];
+      if (!eventTargetActionArray) {
         return;
-
-      void (^removeActionFromTarget)(id <NSCopying> targetKey, SEL action) = ^
-        (id aTarget, SEL theAction)
-        {
-          // Grab the targetActions for this target.
-          NSMutableArray *targetActions = [eventDispatchTable objectForKey:aTarget];
-
-          // Remove action if we have it.
-          if (theAction)
-            [targetActions removeObject:NSStringFromSelector(theAction)];
-          // Or all actions if not.
-          else
-            [targetActions removeAllObjects];
-
-          // If there are no actions left, remove this target entry.
-          if ([targetActions count] == 0)
-          {
-            [eventDispatchTable removeObjectForKey:aTarget];
-
-            // If there are no targets for this event anymore, remove it.
-            if ([eventDispatchTable count] == 0)
-              [_controlEventDispatchTable removeObjectForKey:eventKey];
+      }
+      
+      NSPredicate *filterPredicate = [NSPredicate predicateWithBlock:^BOOL(ASControlTargetAction *_Nullable evaluatedObject, NSDictionary<NSString *,id> * _Nullable bindings) {
```

Comment:
@appleguy Yeah, @Adlai-Holler suggested me that there's a function called `indexOfObjectsPassingTest:` which returns the indexes of requested objects. Then the `removeObjectsAtIndexes:` could be used to remove those items. Do you think calling these have a better performance then creating a predicate and call the `filterUsingPredicate:` function?